From aa5a720056d37cf24924c138a3dbe6dace98e97c Mon Sep 17 00:00:00 2001
From: "H.J. Lu" <hjl.tools@gmail.com>
Date: Thu, 17 Feb 2022 08:10:35 -0800
Subject: [PATCH] string: Add a testcase for wcsncmp with SIZE_MAX [BZ #28755]

Verify that wcsncmp (L("abc"), L("abd"), SIZE_MAX) == 0.  The new test
fails without

commit ddf0992cf57a93200e0c782e2a94d0733a5a0b87
Author: Noah Goldstein <goldstein.w.n@gmail.com>
Date:   Sun Jan 9 16:02:21 2022 -0600

    x86: Fix __wcsncmp_avx2 in strcmp-avx2.S [BZ# 28755]

and

commit 7e08db3359c86c94918feb33a1182cd0ff3bb10b
Author: Noah Goldstein <goldstein.w.n@gmail.com>
Date:   Sun Jan 9 16:02:28 2022 -0600

    x86: Fix __wcsncmp_evex in strcmp-evex.S [BZ# 28755]

This is for BZ #28755.

Reviewed-by: Sunil K Pandey <skpgkp2@gmail.com>

Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/focal/+source/glibc/+bug/2001975
Origin: upstream, https://sourceware.org/git/?p=glibc.git;a=commit;h=aa5a720056d37cf24924c138a3dbe6dace98e97c
---
 string/test-strncmp.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/string/test-strncmp.c b/string/test-strncmp.c
index df7cea4068..1a87f0e73e 100644
--- a/string/test-strncmp.c
+++ b/string/test-strncmp.c
@@ -445,6 +445,18 @@ check2 (void)
 	}
 }
 
+static void
+check4 (void)
+{
+  const CHAR *s1 = L ("abc");
+  CHAR *s2 = STRDUP (s1);
+
+  FOR_EACH_IMPL (impl, 0)
+    check_result (impl, s1, s2, SIZE_MAX, 0);
+
+  free (s2);
+}
+
 int
 test_main (void)
 {
@@ -569,6 +581,7 @@ test_main (void)
   check1 ();
   check2 ();
   check3 ();
+  check4 ();
 
   printf ("%23s", "");
   FOR_EACH_IMPL (impl, 0)
-- 
2.39.2

